packages/f-searchbox/src/services/search.services.js

```diff
@@ -56,71 +55,69 @@ const search = ({
  * @param requiredFields
  * @param streetNumber
  * @param index
+ * @param keyboardSuggestionIndex
  * @returns {Promise.<TResult>}
  */
 const selectedSuggestion = (
     service,
     suggestions,
     requiredFields,
     streetNumber,
-    index
+    index,
+    keyboardSuggestionIndex
 ) => {
-    // TODO pass through suggestion index for keyboard behaviour..
-    const suggestion = suggestions[index || 0];
-    
+    const suggestion = suggestions[index || keyboardSuggestionIndex];
+
     // find current suggestion's location information
     const placeId = suggestion.place_id || suggestion.placeId;
     const streetNumberFormatted = streetNumber.trim();
-    
+
     return service.getLocationDetails(placeId)
         .then(location => service.getLocationPostcode(location, streetNumberFormatted))
         .then(location => {
             location.streetNumber = location.streetNumber || streetNumberFormatted;
             location.suggestion = suggestion.description;
-            
+
             const missingFields = requiredFields.reduce((allFields, field) => (
                 location[field]
                     ? allFields
                     : allFields.concat(field)
             ), []);
-        
-        const streetNumberRequired = requiredFields.includes('streetNumber') && !location.streetNumber;
-        
-        if (!missingFields.length) {
-            const payload = {
-                ...location,
-                city: location.locality,
-                houseNo: location.streetNumber,
-                where: location.postcode
-            };
 
-            search({}, payload);
-            return payload;
-        } else if (
+            const streetNumberRequired = requiredFields.includes('streetNumber') && !location.streetNumber;
+
+            if (!missingFields.length) {
+                const payload = {
+                    ...location,
+                    city: location.locality,
+                    houseNo: location.streetNumber,
+                    where: location.postcode
+                };
+
+                search({}, payload);
+                return payload;
+            } else if (
             // if street required but no street number available
             // display street number field
-            location.street
+                location.street
             && requiredFields.includes('street')
             && missingFields.includes('streetNumber')
```

Comment:
I think it reads cleaner if the indentation is aligned, if the linter doesn't complain?
```suggestion
                && requiredFields.includes('street')
                && missingFields.includes('streetNumber')
```

Reply:
Fixed